### ISSUE TITLE
TelegramBot provider

### ISSUE THREAD

**Issue by csharpfritz** [FritzAndFriends/TagzApp#40]
Surface questions from a Telegram chat room

**Comment by kasib-94** [comment]
Should i here create new ClassLibrary in 2.Providers ? 
I would like to try this task, but i think i would need few more words about how it should be done.

**Comment by csharpfritz** [comment]
No, this should be 1 class library named ‘TagzApp.Providers.Telegram’ that implements the interfaces for a provider. On Jul 29, 2023, at 07:14, kasib-94 ***@***.***> wrote:﻿Re: [FritzAndFriends/TagzApp] TelegramBot provider (Issue #40)Should i here create new ClassLibrary in 2.Providers ?I would like to try this task, but i think i would need few more words about how it should be done.—Reply to this email directly, view it on GitHub or unsubscribe.You are receiving this email because you authored the thread.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by kasib-94** [comment]
Yeah i think you can assign it to me, ill give it a try

**Comment by b4ux1t3** [comment]
Writing this before I've had a chance to actually look at any of the code, but there are two main ways to interact with the Telegram API:

The easiest, but most onerous on resources, is to poll the API every once in a while to grab new messages, and then act on them.

The other way is with a web hook. You subscribe to the types of messages you'd like to receive, and the telegram API makes a web request to some known endpoint that you then monitor and react to.

So, I guess the question is: do we have a preference for which of these we'd like to include, or maybe figure out some way to make it configurable?

That's a question I may be able to answer by looking at all the code first, I just figured it was a good idea to put these thoughts down here.
